Classes/Controller/BackendController.php

```diff
@@ -226,9 +228,29 @@ public function redirectToAction(string $node): void
 
         $nodeAddress = NodeAddress::fromJsonString($node);
 
+        $contentRepository = $this->contentRepositoryRegistry->get($nodeAddress->contentRepositoryId);
+
+        $nodeInstance = $contentRepository->getContentGraph($nodeAddress->workspaceName)->getSubgraph(
+            $nodeAddress->dimensionSpacePoint,
+            VisibilityConstraints::withoutRestrictions()
+        )->findNodeById($nodeAddress->aggregateId);
+
+        // we always want to redirect to the node in the base workspace.
+        $workspace = $contentRepository->getWorkspaceFinder()->findOneByName($nodeAddress->workspaceName);
+
+        $nodeAddressInBaseWorkspace = NodeAddress::create(
+            $nodeAddress->contentRepositoryId,
+            $workspace->baseWorkspaceName ?? WorkspaceName::forLive(),
```

Comment:
If the selected workspace has no base workspace – shouldn't that be an error?

Reply:
i this is like one of these `if ($siteNode) {}` calls that are just there to prevent an error than to run into a fatal error.
If some logic is changed that we for example work on the live workspace directly we should also still be able to preview it ... but i think youre right ... for that case $workspace->baseWorkspaceName ?? $workspace->workspaceName should be enough;)

Reply:
Adjusted:)